app/views/incomeSources/manage/ManageSelfEmployment.scala.html

```diff
@@ -0,0 +1,176 @@
+    if (currentMethod == "Q") {
+        "annual"
+    } else {
+        "quarterly"
+    }
+}
+
+@taxYearValue(taxYear: Int) = @{
+    s"${taxYear.toString}-${(taxYear + 1).toString}"
+}
+
+@extractOption(content: Option[String]) = @{
+    content match {
+        case Some(x) => x
+        case None => messages("incomeSources.generic.unknown")
+    }
+}
+
+@mainTemplate(
+    pageTitle = getMessage("heading"),
+    backUrl = Some(backUrl),
+    useFallbackBackLink = true,
+    isAgent = isAgent,
+    btaNavPartial = user.btaNavPartial) {
+
+    @hmrcPageHeading(PageHeading(
+        text = getMessage("heading"),
+        headingClasses = Some("govuk-heading-l"),
+        section = Some(getMessage("sole-trader-section")),
+        captionClasses = Some("govuk-caption-l")
+    ))
+
+    <dl class="govuk-summary-list" id="refund-to-taxpayer-table">
+        <div class="govuk-summary-list__row">
+            <dt class="govuk-summary-list__key">@getMessage("business-name")</dt>
+            <dd class="govuk-summary-list__value" id="business-name">@{
+                extractOption(viewModel.tradingName)
+            }</dd>
+        </div>
+        <div class="govuk-summary-list__row">
+            <dt class="govuk-summary-list__key">@getMessage("business-address")</dt>
+            <dd class="govuk-summary-list__value" id="business-address">@{
+                viewModel.address match {
+                    case Some(x) => HtmlFormat.raw(x)
+                    case None => messages("incomeSources.generic.unknown")
+                }
+            }</dd>
+        </div>
+        <div class="govuk-summary-list__row">
+            <dt class="govuk-summary-list__key">@getMessage("date-started")</dt>
+            <dd class="govuk-summary-list__value" id="business-date-started">@{
+                viewModel.tradingStartDate match {
+                    case Some(x) => x.toLongDate
+                    case None => messages("incomeSources.generic.unknown")
+                }
+            }</dd>
+        </div>
+        <div class="govuk-summary-list__row">
+            <dt class="govuk-summary-list__key">@getMessage("accounting-method")</dt>
+            <dd class="govuk-summary-list__value" id="business-accounting-method">@{
+                viewModel.businessAccountingMethod match {
+                    case Some(x) => x match {
+                        case value if value == "cash" => getMessage("cash-accounting")
+                        case value if value == "accruals" => getMessage("traditional-accounting")
+                        case _ => throw new MissingFieldException("Unexpected value for businessAccountingMethod field")
+                    }
+                    case None => messages("incomeSources.generic.unknown")
+                }
+            }</dd>
+        </div>
+        @if(viewModel.itsaHasMandatedOrVoluntaryStatusCurrentYear.getOrElse(throw MissingFieldException("Missing itsaHasMandatedOrVoluntaryStatusCurrentYear field"))
```

Comment:
pls avoid throwing exception in the views

Reply:
this should never be empty so can I just do .get?

Reply:
in case in this context you are 100% sure that value would be provided, you can call .get; 
but its worth to check if this field really need to be Optional  ?

Reply:
yes I removed the optional